runsc/container/container_test.go

```diff
@@ -3127,3 +3129,160 @@ func TestOverlayByMountAnnotation(t *testing.T) {
 		t.Fatalf("overlay filestore at %q was not deleted after container.Destroy()", filestoreFile)
 	}
 }
+
+// TestMountEROFS checks that the checksums from the target directory in container
+// are identical with the ones from the source directory on host.
+func TestMountEROFS(t *testing.T) {
+	// Create a temporary directory with some random files in it, which will
+	// be used as the source directory to create the EROFS images.
+	sourceDir, err := ioutil.TempDir(testutil.TmpDir(), "erofs-source")
+	if err != nil {
+		t.Fatalf("ioutil.TempDir() failed: %v", err)
+	}
+
+	// Create a big directory.
+	bigDir, err := ioutil.TempDir(sourceDir, "erofs_test_")
+	if err != nil {
+		t.Fatalf("ioutil.TempDir() failed: %v", err)
+	}
+	for i := 0; i < 1000; i++ {
+		// Create a random regular file.
+		tempFile, err := ioutil.TempFile(bigDir, "erofs_test_")
+		if err != nil {
+			t.Fatalf("ioutil.TempFile() failed: %v", err)
+		}
+		// Write some random data into the regular file.
+		if _, err := fmt.Fprint(tempFile, rand.Int()); err != nil {
+			t.Fatalf("fmt.Fprint() failed: %v", err)
+		}
+		name := tempFile.Name()
+		// Create a symlink that points to the regular file.
+		if err := os.Symlink(name, name+"@"); err != nil {
+			t.Fatalf("os.Symlink() failed: %v", err)
+		}
+		tempFile.Close()
+	}
```

Comment:
I created a binary sometime back to do exactly this. Could you just use that? See [`test/cmd/test_app/main.go`](https://github.com/google/gvisor/blob/0e18a826d91d20409ef32cd98c9b0512aadeb93d/test/cmd/test_app/main.go#L63-L114).

See an example usage of this binary in a container test [here](https://github.com/google/gvisor/blob/0e18a826d91d20409ef32cd98c9b0512aadeb93d/runsc/container/multi_container_test.go#L2337).

You can add a new optional flag that tell the directory under which to create the filesystem tree and pass `sourceDir` as that path. And instead of executing the binary inside the container (like the example does), just run it outside.

Reply:
Cool. This tool is just what I wanted. I will use it. Thanks!